src/dataflow/src/storage/render/debezium.rs

```diff
@@ -118,6 +121,304 @@ where
+
+                        while let Some((data_cap, refmut_data)) = data_input.next() {
+                            // RefOrMut doesn't let us drain directly into iterator
+                            refmut_data.swap(&mut data);
+                            let data_cap = data_cap.retain();
+                            data_buffer.extend(data.drain(..).map(|r| (r, data_cap.clone())));
+                        }
+                        while let Some((result, data_cap)) = data_buffer.get(0).cloned() {
+                            let key = match result.key.transpose() {
+                                Ok(key) => key,
+                                Err(err) => {
+                                    output.session(&data_cap).give((
+                                        Err(err.into()),
+                                        *data_cap.time(),
+                                        1,
+                                    ));
+                                    let _ = data_buffer.pop_front();
+                                    continue;
+                                }
+                            };
+                            let value = match result.value {
+                                Some(Ok(value)) => value,
+                                Some(Err(err)) => {
+                                    output.session(&data_cap).give((
+                                        Err(err.into()),
+                                        *data_cap.time(),
+                                        1,
+                                    ));
+                                    let _ = data_buffer.pop_front();
+                                    continue;
+                                }
+                                None => {
+                                    let _ = data_buffer.pop_front();
+                                    continue;
+                                }
+                            };
+
+                            let tx_id = match value.iter().nth(transaction_idx).unwrap() {
+                                Datum::List(l) => {
+                                    let tx_id = l.iter().nth(tx_metadata_description.data_transaction_id_idx).unwrap().unwrap_str().to_owned();
+                                    let tx_time: Timestamp = match tx_mapping.get(&tx_id) {
+                                        Some(time) => *time,
+                                        None => break,
+                                    };
+                                    Some((tx_id, tx_time))
+                                },
+                                Datum::Null => None,
+                                _ => panic!("Previously validated to be nullable list"),
+                            };
+
+                            // We're committed to processing it if we have a tx_time
+                            let _ = data_buffer.pop_front();
+
+                            let partition_dedup = dedup_state
```

Comment:
Can you leave a TODO comment here about re-using the dedup operator we already have?